UPGRADE-3.4.md

```diff
@@ -114,6 +114,40 @@ FrameworkBundle
    class has been deprecated and will be removed in 4.0. Use the
    `Symfony\Component\Translation\DependencyInjection\TranslatorPass` class instead.
 
+HttpKernel
+----------
+
+ * The `getCacheDir()` method of your kernel should return the non-empty value
+   from `Kernel::getWarmupDir()` when one exists; not doing so is deprecated.
```

Comment:
I don't think it's possible without doing this. That's the feature of getter injection: it makes things really immutable, unless a specific mechanism exists to work around - which is what this achieves.